src/index.ts

```diff
@@ -111,42 +129,42 @@ function deriveKeypair(seed, options) {
   return keypair
 }
 
-function getAlgorithmFromKey(key) {
+function getAlgorithmFromKey(key): 'ed25519' | 'ecdsa-secp256k1' {
   const bytes = hexToBytes(key)
   return bytes.length === 33 && bytes[0] === 0xed
     ? 'ed25519'
     : 'ecdsa-secp256k1'
 }
 
-function sign(messageHex, privateKey) {
+function sign(messageHex, privateKey): string {
   const algorithm = getAlgorithmFromKey(privateKey)
   return select(algorithm).sign(hexToBytes(messageHex), privateKey)
 }
 
-function verify(messageHex, signature, publicKey) {
+function verify(messageHex, signature, publicKey): boolean {
   const algorithm = getAlgorithmFromKey(publicKey)
   return select(algorithm).verify(hexToBytes(messageHex), signature, publicKey)
 }
 
-function deriveAddressFromBytes(publicKeyBytes: Buffer) {
+function deriveAddressFromBytes(publicKeyBytes: Buffer): string {
   return addressCodec.encodeAccountID(
     utils.computePublicKeyHash(publicKeyBytes),
   )
 }
 
-function deriveAddress(publicKey) {
-  return deriveAddressFromBytes(hexToBytes(publicKey))
+function deriveAddress(publicKey): string {
+  return deriveAddressFromBytes(Buffer.from(hexToBytes(publicKey)))
 }
 
-function deriveNodeAddress(publicKey) {
+function deriveNodeAddress(publicKey): string {
   const generatorBytes = addressCodec.decodeNodePublic(publicKey)
   const accountPublicBytes = accountPublicFromPublicGenerator(generatorBytes)
   return deriveAddressFromBytes(accountPublicBytes)
 }
 
 const { decodeSeed } = addressCodec
 
-module.exports = {
+export = {
```

Comment:
Is it worth switching this from exporting an object to putting the `export`s next to the functions they export? I think that's a bit more conventional.

Reply:
I think this is similar to a default export for CommonJS. I am fine with putting the `export`s next to the functions, but only if it allows for the `import keypairs from 'ripple-keypairs'` syntax that ripple-lib currently uses for importing ripple-keypairs.